src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -92,6 +92,25 @@ public static string GetDirectoryName(string path)
             return null;
         }
 
+        public static ReadOnlySpan<char> GetDirectoryName(ReadOnlySpan<char> path)
+        {
+            if (PathInternal.IsEffectivelyEmpty(new string(path.ToArray())))
+                throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
+
+            path = PathInternal.NormalizeDirectorySeparators(new string(path.ToArray())).AsReadOnlySpan();
+            int root = PathInternal.GetRootLength(new string(path.ToArray()));
+
+            int i = path.Length;
+            if (i > root)
+            {
+                while (i > root && !PathInternal.IsDirectorySeparator(path[--i]))
+                    ;
+                return path.Slice(0, i);
+            }
+
+            return null;
```

Comment:
You're returning a ReadOnlySpan which is an struct, can't be null, instead just return an empty span.